cmake/everest-generate.cmake

```diff
@@ -494,7 +499,82 @@ function (ev_add_py_module MODULE_NAME)
             add_subdirectory(${MODULE_PATH})
         endif()
     else()
-        message(WARNING "C++ module ${MODULE_NAME} does not exist at ${MODULE_PATH}")
+        message(WARNING "Python module ${MODULE_NAME} does not exist at ${MODULE_PATH}")
+        return()
+    endif()
+
+    # FIXME (aw): this will override EVEREST_MODULES, might not what we want
+    set_property(
+        GLOBAL
+        PROPERTY EVEREST_MODULES ${EVEREST_MODULES}
+    )
+endfunction()
+
+function (ev_add_rs_module MODULE_NAME)
+    set(EVEREST_MODULE_INSTALL_PREFIX "${CMAKE_INSTALL_LIBEXECDIR}/everest/modules")
+
+    set(MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/${MODULE_NAME}")
+    set(MODULE_BUILD_PATH "${CMAKE_CURRENT_BINARY_DIR}/${MODULE_NAME}")
+
+    if(IS_DIRECTORY ${MODULE_PATH})
+        if(NOT ${EVEREST_ENABLE_RS_SUPPORT})
+            message(STATUS "Excluding Rust module ${MODULE_NAME} because EVEREST_ENABLE_RS_SUPPORT=${EVEREST_ENABLE_RS_SUPPORT}")
+            return()
+        elseif("${MODULE_NAME}" IN_LIST EVEREST_EXCLUDE_MODULES)
+            message(STATUS "Excluding module ${MODULE_NAME}")
+            return()
+        elseif(EVEREST_INCLUDE_MODULES AND NOT ("${MODULE_NAME}" IN_LIST EVEREST_INCLUDE_MODULES))
+            message(STATUS "Excluding module ${MODULE_NAME}")
+            return()
+        else()
+            message(STATUS "Setting up Rust module ${MODULE_NAME}")
+
+            find_program(
+                    RSYNC
+                    rsync
+                    REQUIRED
+            )
+
+            # Copy source of module ${MODULE_NAME} to build dir
+            execute_process(
+                COMMAND ${RSYNC} -avq --delete "${MODULE_PATH}/" "${MODULE_BUILD_PATH}"
+            )
+
+            find_program(
+                SED
+                sed
+                REQUIRED
+            )
+
+            execute_process(
+                COMMAND ${SED} -i "/everestrs = /c\\everestrs = { path = \"${everest-framework_SOURCE_DIR}/everestrs/everestrs\" }" "${MODULE_BUILD_PATH}/Cargo.toml"
+            )
+
+            find_program(
+                CARGO
+                cargo
+                REQUIRED
+            )
+
+            add_custom_target(
+                ${MODULE_NAME} ALL
+                COMMAND ${CMAKE_COMMAND} -E env EVEREST_RS_FRAMEWORK_SOURCE_LOCATION="${everest-framework_SOURCE_DIR}" EVEREST_RS_FRAMEWORK_BINARY_LOCATION="${everest-framework_BINARY_DIR}" ${CARGO} build
+                WORKING_DIRECTORY ${MODULE_BUILD_PATH}
```

Comment:
Here I would use the generator expressions, to directly link to the correct library

Reply:
I do not understand what this means.

Reply:
In cmake you have generator expressions, which can give you information about targets and other things.  Say for example you have the library `everest::framework` (which resolves to a target), then you can get the full path to it via `$<TARGET_FILE:everest::framework>` instead of using *guesses*

Reply:
I defer here, not sure how to implement that.